src/basic/rlimit-util.c

```diff
@@ -55,6 +61,39 @@ int setrlimit_closest(int resource, const struct rlimit *rlim) {
         return 0;
 }
 
+int setpriority_closest(int nice) {
+        int current, limit;
+        struct rlimit highest;
+
+        // Try to set requested nice level
+        if (setpriority(PRIO_PROCESS, 0, nice) == 0)
+                return 0;
+
+        // Permission failed
+        if (!IN_SET(errno, EPERM, EACCES))
+                return -errno;
+
+        errno = 0;
+        current = getpriority(PRIO_PROCESS, 0);
+        if (errno)
+                return -errno;
+
+        if (getrlimit(RLIMIT_NICE, &highest) < 0)
+                return -errno;
+
+        limit = rlimit_to_nice(highest.rlim_cur);
+
+        // We are already nicer than limit allows us
+        if (current < limit)
```

Comment:
maybe a log_debug() line here that says "can't raise nice level, permissions and resource limit don't allow it"

Reply:
But... it needs to be log_unit_debug, no? And thus I need to pass the pointer to the unit to this function too. Since this happens during unit setup, and regular logging with go nowhere, no? This is how I understood the logging refactor during unit spawning, no?